### PR TITLE
Updated to 0.5.7

### PR DESCRIPTION
**Pull request by hmaarrfk**
The devel libraries were packaged with the original program.
I also removed the static libraries and put them in their own package.
Enabled usb, packet-socket and fast-heartbeat (don't know what this one does, but it says fast).

### CODE REVIEW

File: aravis.spec [hunk 1]
```diff
@@ -128,10 +122,45 @@ fi
...
 
 %changelog
+* Tue Feb 21 2017 Mark Harfouche <mark.harfouche@gmail.com> 0.5.7
+- New upstream release
+- Enabled usb support
+- I didn't know where to put
+%{_datadir}/%{fullname}/arvviewer.h
+%{_datadir}/%{fullname}/arvviewertypes.h
+- Enabled packet-socket
+- Enabled fast-heartbeat
```
Comment (EmmanuelP) [inline aravis.spec]: Please remove fast-heartbeat, it is an advanced feature, and it even should not be a compile time choice.

File: aravis.spec [hunk 2]
```diff
@@ -128,10 +122,45 @@ fi
...
+%{_datadir}/appdata/arv-viewer.appdata.xml
+
 %files gstreamer1
 %{_libdir}/gstreamer-1.0/*
 
 %changelog
+* Tue Feb 21 2017 Mark Harfouche <mark.harfouche@gmail.com> 0.5.7
+- New upstream release
+- Enabled usb support
+- I didn't know where to put
```
Comment (EmmanuelP) [inline aravis.spec]: Don't install these headers, they are just used for viewer compilation.

File: aravis.spec [hunk 3]
```diff
@@ -20,16 +20,21 @@ BuildRequires:	pkgconfig(gtk+-3.0)
...
 BuildRequires:	pkgconfig(gstreamer-base-1.0) >= 1.0
 BuildRequires:	pkgconfig(gstreamer-app-1.0)
+BuildRequires:	pkgconfig(libusb-1.0)
+BuildRequires:	pkgconfig(libcap-ng)
+BuildRequires:	pkgconfig(audit)
 
 Source10:	aravis.png
 
-%global fullname %{name}-0.4
+# I don't know why version 0.5.7 is actually 0.6
```
Comment (EmmanuelP) [inline aravis.spec]: 0.6 is the API version, and 0.5.7 is the version number of the development series. The stable series will have 0.6.x versions.

**Review by EmmanuelP — COMMENTED** [review]
Thanks for this update. As you seem to know rpm better than me, would it be possible to make aravis and aravis-devel rpms parallel installable with the 0.4 series ?

File: aravis.spec (hunk 3 above)
Comment (hmaarrfk) [reply aravis.spec]: I got that after a while. Seems to be the default convention for Gnome Software

File: aravis.spec (hunk 2 above)
Comment (hmaarrfk) [reply aravis.spec]: I guess then that this is a bug because the viewer seems to install them on it's own.

**Comment by hmaarrfk** [comment]
I think you can have multiple versions of the libraries, but I think no matter what, the viewer would conflict.

I think we would have to name them aravis-0.4 and aravis-0.6 in the package names, kinda like what gstreamer did. It is ugly IMO.

**Comment by EmmanuelP** [comment]
> I think you can have multiple versions of the libraries, but I think no matter what, the viewer would conflict.

I'm OK with that. What I think is important is to allow installation of both stable and development versions of the aravis libraries, that lets the user tests the latest version without breaking something developed using the stable version.

> I think we would have to name them aravis-0.4 and aravis-0.6 in the package names, kinda like what gstreamer did. It is ugly IMO.

Is it possible to indicate a package is in conflict or replace another one, because an aravis-0.4 should replace a installed aravis package.

**Comment by EmmanuelP** [comment]
By the way, I have pushed to aravis master a patch that removes the installation of viewer headers.

**Comment by hmaarrfk** [comment]
Yes it is possible to indicate a package superseeds an other. Lets go with that solution. Your library is actually aravis-0.4 and not simply aravis. In fact, I think it installs in directories that have the version number in them.

I'll create a new branch for the rename
https://fedoraproject.org/wiki/Upgrade_paths_%E2%80%94_renaming_or_splitting_packages

**Comment by EmmanuelP** [comment]
Thanks Mark.

I have pushed your spec files to master. Instead of using branches, I have opted for separate spec files.

There is still a conflict between the two aravis-0.x-viewer rpms, as they install the files in the same place. I'm not sure what to do about it. While I find important to be able to have both libraries installed at the same time, it seems less an issue to force the use of the latest viewer.

**Comment by hmaarrfk** [comment]
cool!

**Pull request closed without merging**